### DISCUSSION TITLE
Event Dispatcher

### DISCUSSION THREAD

**Discussion by mhgtlc** [pimcore/pimcore#11913] [Q&A / Support]
Is there an option do disable dispatching of element post save events for `isAutoSave` or `saveVersionOnly` cases?

**Comment by dvesh3** [comment]
you can return early in event listeners:
```php
if ($e->hasArgument('saveVersionOnly') || $e->hasArgument('autoSave')) {
    return;
}
```

**Reply by mhgtlc** [reply]
Thank you for your reply! Actually we did so, however we do have multiple event listeners, and we have to apply this check to all of them which is not convenient. May be it would be a good idea to add a config for event listeners to be able to disable event dispatching on `autoSave` and `saveVersionOnly`, what do you think?

**Reply by dvesh3** [reply]
Maybe even better to have a generic config which allows to disable any event. I would like to see a PR 😉
